apps/admin-gui/src/app/shared/components/entityless-attribute-keys-list/entityless-attribute-keys-list.component.ts

```diff
@@ -121,11 +120,13 @@ export class EntitylessAttributeKeysListComponent implements OnChanges, OnInit {
     const rec = ['', this.attDef as Attribute] as [string, Attribute];
     rec[1].value = undefined;
     this.records.push(rec);
-    this.ngOnChanges(undefined);
+    this.dataSource.data = this.records;
+    this.setDataSource();
     this.selection.clear();
     this.selection.select(rec);
     this.isAddButtonDisabled = true;
-    this.paginator.lastPage();
+    // .lastPage() has to be delayed, otherwise it wont go to last page if number of records modulo page size is 1
+    setTimeout(() => this.paginator.lastPage(), 100);
```

Comment:
I think it makes sense to add the new record to the start of the list. Also, I am a little worried about this solution which is using the timeout.

Reply:
Done, there's no timeout anymore, and added attribute is on top (although after save backend puts him at the end)